### ISSUE TITLE
Method Chaining in Tensorflow

### ISSUE THREAD

**Issue by najibghadri** [tensorflow/tensorflow#37311]
I have posted a question about this in stackoverflow. https://stackoverflow.com/questions/60528238/why-no-method-chaining-in-tensorflow

Aka. named parameter idiom. I did a simple Nearest Neighbor algorithm in TF and I ended up writing the whole NN algorithm in one line. You can argue that I should take it apart, but I think this happens often anyways (in coding in general too):

```
tf.math.argmin(tf.math.reduce_sum(tf.math.abs(tf.math.subtract(Xtrain_set, Xtest_row)), axis=1), output_type=tf.int32)

# returns the minimum train set indices for each X predictable row based on L1 (in a for cycle) but this is not important
```

It could be much more readable and simple to write:

**`tf.subtract(Xtrain_set, Xtest_row).abs().reduce_sum(axis=1).argmin(output_type=tf.int32)
`**

I think it really follows the "Tensor-flow" thinking too.
Thanks.

**Comment by bersbersbers** [comment]
I have a similar idea for the Keras interface: instead of
```
model = vgg16.VGG16()
model.compile(loss="mse")
model.fit(...)
```

it would be nice to be able to do
```
vgg16.VGG16().compile(loss="mse").fit(...)
```

And I think all it takes in this examples is a simple `return(self)` here:
https://github.com/tensorflow/tensorflow/blob/b36436b087bd8e8701ef51718179037cccdfc26e/tensorflow/python/keras/engine/training.py#L462-L470
https://github.com/tensorflow/tensorflow/blob/b36436b087bd8e8701ef51718179037cccdfc26e/tensorflow/python/keras/engine/training.py#L555-L557

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
